src/Custom/Microsoft.Extensions.AI/OpenAIChatClient.cs

```diff
@@ -0,0 +1,760 @@
+                {
+                    result.StoredOutputEnabled = storeOutputEnabled;
+                }
+
+                if (additionalProperties.TryGetValue(nameof(result.TopLogProbabilityCount), out int topLogProbabilityCountInt))
+                {
+                    result.TopLogProbabilityCount = topLogProbabilityCountInt;
+                }
+            }
+
+            if (options.Tools is { Count: > 0 } tools)
+            {
+                foreach (AITool tool in tools)
+                {
+                    if (tool is AIFunction af)
+                    {
+                        result.Tools.Add(ToOpenAIChatTool(af));
+                    }
+                }
+
+                if (result.Tools.Count > 0)
+                {
+                    switch (options.ToolMode)
+                    {
+                        case NoneChatToolMode:
+                            result.ToolChoice = ChatToolChoice.CreateNoneChoice();
+                            break;
+
+                        case AutoChatToolMode:
+                        case null:
+                            result.ToolChoice = ChatToolChoice.CreateAutoChoice();
+                            break;
+
+                        case RequiredChatToolMode required:
+                            result.ToolChoice = required.RequiredFunctionName is null ?
+                                ChatToolChoice.CreateRequiredChoice() :
+                                ChatToolChoice.CreateFunctionChoice(required.RequiredFunctionName);
+                            break;
+                    }
+                }
+            }
+
+            if (options.ResponseFormat is ChatResponseFormatText)
+            {
+                result.ResponseFormat = OpenAI.Chat.ChatResponseFormat.CreateTextFormat();
+            }
+            else if (options.ResponseFormat is ChatResponseFormatJson jsonFormat)
+            {
+                result.ResponseFormat = jsonFormat.Schema is { } jsonSchema ?
+                    OpenAI.Chat.ChatResponseFormat.CreateJsonSchemaFormat(
+                        jsonFormat.SchemaName ?? "json_schema",
+                        BinaryData.FromBytes(
+                            JsonSerializer.SerializeToUtf8Bytes(jsonSchema, ChatClientJsonContext.Default.JsonElement)),
+                        jsonFormat.SchemaDescription,
+                        jsonSchemaIsStrict: true) :
+                    OpenAI.Chat.ChatResponseFormat.CreateJsonObjectFormat();
+            }
+        }
+
+        return result;
+    }
+
+    /// <summary>Converts an Extensions function to an OpenAI chat tool.</summary>
+    private static ChatTool ToOpenAIChatTool(AIFunction aiFunction)
+    {
+        // Default strict to true, but allow to be overridden by an additional Strict property.
+        bool strict =
+            !aiFunction.AdditionalProperties.TryGetValue("Strict", out object? strictObj) ||
+            strictObj is not bool strictValue ||
+            strictValue;
```

Comment:
Can you coordinate with @eiriktsarpalis to make sure his changes in https://github.com/dotnet/extensions/pull/6285 don't get lost here?